collector-kinesis/src/main/java/zipkin/collector/kinesis/KinesisCollector.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.collector.kinesis;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.v2.IRecordProcessorFactory;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.KinesisClientLibConfiguration;
+import com.amazonaws.services.kinesis.clientlibrary.lib.worker.Worker;
+import zipkin.collector.Collector;
+import zipkin.collector.CollectorComponent;
+import zipkin.collector.CollectorMetrics;
+import zipkin.collector.CollectorSampler;
+import zipkin.internal.Util;
+import zipkin.storage.StorageComponent;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+public final class KinesisCollector implements CollectorComponent, Closeable {
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder implements CollectorComponent.Builder {
+
+        Collector.Builder delegate = Collector.builder(KinesisCollector.class);
+
+        AWSCredentialsProvider credentialsProvider;
+        String appName;
+        String streamName;
+
+        @Override
+        public Builder storage(StorageComponent storageComponent) {
+            delegate.storage(storageComponent);
+            return this;
+        }
+
+        @Override
+        public Builder metrics(CollectorMetrics collectorMetrics) {
+            delegate.metrics(Util.checkNotNull(collectorMetrics, "metrics").forTransport("kinesis"));
+            return this;
+        }
+
+        @Override
+        public Builder sampler(CollectorSampler collectorSampler) {
+            delegate.sampler(collectorSampler);
+            return this;
+        }
+
+        public Builder credentialsProvider(AWSCredentialsProvider credentialsProvider) {
+            this.credentialsProvider = credentialsProvider;
+            return this;
+        }
+
+        public Builder appName(String appName) {
+            this.appName = appName;
+            return this;
+        }
+
+        public Builder streamName(String streamName) {
+            this.streamName = streamName;
+            return this;
+        }
+
+        @Override
+        public KinesisCollector build() {
+            return new KinesisCollector(this);
+        }
+
+        Builder() {
+        }
+    }
+
+    private Collector collector;
```

Comment:
nit: final all the things and double space indent (square formatter will do the indent stuff)